### DISCUSSION TITLE
I can't find different namespaces, why?

### DISCUSSION THREAD

**Discussion by bezenaws** [GoogleCloudPlatform/microservices-demo#783] [General]
Hi,

I am new to K8s and trying to deploy this demo-project to AWS platform. I have a quick question, couple of days back I saw there was separate namespaces created for each micro services but seems like everything will be be deployed to default namespace. 

But I am under impression, that if we have separate namespaces for each module, it will be good. Just my thought. 

I try to find the version that has different namespaces but was not able to find. 

Can someone please help me understand why not have few namespaces?

**Comment by NimJay** [comment]
**Summary:** Only the `default` Namespace is used because we want to keep the Kubernetes manifests as simple as possible and we don't (yet) tell a story about the fictitious "team" working on Online Boutique.

Hi @bezenaws! 

Thanks for starting the very first discussion in this repo! 🥳

You're right, all microservices in this repository are deployed into the `default` Kubernetes namespace.

You might have already seen this article: [Kubernetes best practices: Organizing with Namespaces](https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-organizing-with-namespaces#:~:text=Depending%20on%20your%20situation%2C%20you%20can%20adopt%20the%20relevant%20Namespace%20strategy.). It talks a bit about Namespace best practices — for 3 different organization sizes.

My understanding is that Namespace are mostly used to organize **teams** and the different **_types_ of deployments** (e.g., production vs. testing vs. development). Online Boutique's story doesn't (yet) say anything about the fictitious development teams working on each microservice (e.g., how many development "teams" exist, which microservice(s) does each "team" own, etc.), so it's hard to create meaningful Namespaces.

But I'd say the primary reason for using the `default` Namespace for all microservices is to keep Online Boutique's Kubernetes manifests (and `kubectl` commands in tutorials using Online Boutique) as simple as possible. 

Hope this helps! :)
Nim

P.S. - I'm no Namespace expert — so let me know if I'm missing something or if you have more suggestions. :)

**Comment by george848** [comment]
If you are referring to namespaces in the context of programming, it's possible that the programming language or framework you are using does not have built-in support for namespaces. Not all programming languages or frameworks utilize namespaces as a means of organizing code[.](https://apkvenom.com/)

Namespaces are a feature found in several programming languages, such as C++, C#, and PHP. They provide a way to group related classes, functions, or variables together and avoid naming conflicts. However, not all languages have this concept, and some may use alternative mechanisms for organizing code, such as modules, packages, or modules.

If you could provide more details about the programming language or framework you are using, I could give you more specific information on how code organization is typically handled in that particular context.
